.github/workflows/main.yaml

```diff
@@ -33,10 +33,28 @@ jobs:
       workload_identity_provider: ${{ vars.GUNDI_INTEGRATIONS_WORKLOAD_IDENTITY_PROVIDER}}
       service_account: ${{ vars.GUNDI_INTEGRATIONS_SERVICE_ACCOUNT }}
 
+  run_unit_tests:
+    runs-on: ubuntu-latest
+    needs: [vars, build]
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10' 
```

Comment:
Optional: cache pip dependencies.
https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies

Reply:
@daniefdz @davidovichmarcos 
I also see that the build step fails in the template repo, and that blocks the `run_unit_tests` step to run. We don't really need to run the build step in the template repo, or it wouldn't matter if we do and it fails, but I understand that it's there so we use it in specific integrations. What do you think if we invert the dependency? So unit tests run first, and then we do the build. I think that would work for both the template repo and integrations repos.

Reply:
right, it makes sense to move the tests before the build

Reply:
done!